### PR TITLE
Whitelist abbr tags with title

### PR DESCRIPTION
**Pull request by sbeaumontweb**
## Summary

Adding abbreviation tags to the whitelist for tags allowed to have a title attribute.

## Motivation and Context

Fixing https://github.com/bbc/bbc-a11y/issues/281

## How Has This Been Tested?

Added a new test to the multi-tiers that covers the scenario of `<abbr>` with a title existing within a page.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I've added tests for my code
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by sbeaumontweb** [comment]
Hi,

Sure - let's discuss it in the relevant issue? https://github.com/bbc/bbc-a11y/issues/281
This keeps the discussion where the problem was raised 👍

**Comment by EmmaJP** [comment]
@JamieKnightBBC as you know the <abbr> tag is intended to be used for all types of abbreviation, when appropriate, and using a title attribute to expand or define is encouraged (https://www.w3.org/TR/WCAG20-TECHS/H28.html). This does not preclude the need for the expanded version to be used earlier or elsewhere in the content to assist understanding, or the addition of an aria-label attribute to improve pronunciation for screen reader users when needed.

MAG states "Tooltips must not repeat link text or other alternatives" meaning text alternatives. https://www.bbc.co.uk/guidelines/futuremedia/accessibility/mobile/text-equivalents/tooltips-and-supplementary-information. A title attribute on an <abbr> element should not flag up as a bbc-a11y accessibility fail because it is not necessarily critical information, nor is it repeating the text or alternative text of the element.

That said...

@sbeaumontweb not all abbreviations require <abbr>, which can break screen reader flow, eg. commonly known acronyms like BBC and NATO are usually fine without them. <abbr> tags are not interactive by default, so a keyboard or switch user cannot trigger the tooltip that some mouse users may see if they hover for long enough. And most screen reader users won't hear the title attribute, for a variety of reasons.

https://stackoverflow.com/questions/32888344/abbr-element-aria-label-or-title-attribute is an interesting read. So is https://openinclusion.com/blog/presenting-abbreviations-acronyms-for-screen-reader-users/.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bbc-a11y",
-  "version": "2.2.6",
+  "version": "2.2.7",
```
Comment (EmmaJP) [inline package.json]: @andymsuk should this be included in the commit, or kept for the release?

**Comment by EmmaJP** [comment]
I don't consider a title attribute on an `<abbr>` element to be an accessibility fail, however, that is not the same as implying the title attribute makes the element fully accessible.

Should bbc-a11y be flagging up things like this for closer inspection? Or would such warnings muddy the waters? 

I guess the conversation at #281 might guide thoughts on that.

File: package.json (hunk 1 above)
Comment (andymsuk) [reply package.json]: @EmmaJP - good spot. 
@sbeaumontweb Please can this be removed and we'll do it as part of the release process?

Comment (sbeaumontweb) [reply package.json]: Yep, sorry, complete muscle memory from working on morph modules for so long, haha.

Comment (andymsuk) [reply package.json]: No worries! :)

Comment (andymsuk) [reply package.json]: @sbeaumontweb  
_Bump_

**Comment by EmmaJP** [comment]
@sbeaumontweb thanks for removing the version number change. I'm not sure why the checks are failing. I'm also not sure the conversation around this change has been resolved.

It's a tough one. `title` attributes on a `<abbr>` are not strictly an accessibility fail, but they are also not sufficient to make the content accessible. The [MAG](https://www.bbc.co.uk/accessibility/forproducts/guides/mobile/tooltips-and-supplementary-information) requires tooltips do not repeat the element content, and for HTML the recommended best practice is not to use `title` on elements other than form inputs without explicit labels. Yet, I am aware iframes also use `title` and are in the list of excluded elements for this test for similar reasons.

**Comment by EmmaJP** [comment]
@sbeaumontweb I'm aware sport came up with a different solution for abbreviated names. Do you know if anything should change about this PR? Should it still go ahead? Would this be one for someone else to pick up, or could you fit it in?

**Comment by sbeaumontweb** [comment]
I'm going to close this as no definite decision came out of the relevant issue discussions.

**Pull request closed without merging**